ddtrace/tracer/rules_sampler.go

```diff
@@ -307,6 +343,18 @@ func (rs *traceRulesSampler) enabled() bool {
 	return len(rs.rules) > 0 || !math.IsNaN(rs.globalRate)
 }
 
+func Equals(a, b []SamplingRule) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for i, r := range a {
+		if !r.Equals(&b[i]) {
+			return false
+		}
+	}
+	return true
```

Comment:
Just a question: is permutation a risk? Admittedly, it wouldn't change the end result however

Reply:
It is not a risk. However, I think the method should have a name like `EqualsFalseNegative` to indicate that it is not a strict/accurate `Equals`.

Also added the comment to explain.